plutus-core/plutus-core/test/Evaluation/Spec.hs

```diff
@@ -51,24 +42,71 @@ test_builtinsDon'tThrow =
  `Term`s (which technically doesn't guarantee evaluation success, although it is the case
  with all current builtin functions).
 -}
-prop_builtinsDon'tThrow ::
+test_builtinsDon'tThrow :: TestTree
+test_builtinsDon'tThrow =
+    testGroup
+        "Builtins don't throw"
+        $ fmap
+            (\fun -> testProperty (display fun) $ prop_builtinEvaluation @DefaultFun fun gen f)
+            List.enumerate
+  where
+    gen bn = Gen.choice [genArgsWellTyped bn, genArgsArbitrary bn]
+    f bn args = \case
+        Left e -> do
+            annotate "Builtin function evaluation failed"
+            annotate $ "Function: " <> display bn
+            annotate $ "Arguments: " <> display args
+            annotate $ "Error " <> show e
```

Comment:
The way `annotate` works is that you can do it unconditionally, and it only gets shown on a failure. So you can do something like this:
```
prop_builtinEvaluation :: .... -> Bool -> ...
  ...
  annotate $ "Function: " <> display bn
  annotate $ "Arguments: " <> display args
  annotateShow res
  if expectThrow
  then assertBool "Builtin function evaluation succeeded" (isLeft res)
  else assertBool "Builtin function evaluation failed" (isRight res)
```
Which I think is a bit simpler since you avoid needing a continuation!